packages/plugin/src/rules/require-import-fragment/index.ts

```diff
@@ -90,13 +90,16 @@ export const rule: GraphQLESLintRule = {
           const extractedImportPath = comment.value.match(/(["'])((?:\1|.)*?)\1/)?.[2];
           if (!extractedImportPath) continue;
 
-          const importPath = path.join(path.dirname(filePath), extractedImportPath);
+          const importPath = path.join(
+            path.dirname(filePath),
+            extractedImportPath.replaceAll(/[/\\]/g, path.sep),
+          );
+
           const hasInSiblings = fragmentsFromSiblings.some(
             source => source.filePath === importPath,
           );
           if (hasInSiblings) return;
         }
-
         const fragmentInSameFile = fragmentsFromSiblings.some(
           source => source.filePath === filePath,
```

Comment:
same, `source.filePath` comes from `graphql-config` and contains slashes according to user's OS, and `filePath` is ESLint's `context.filename` which also contains slashed according user's OS